src/gui/traymenu.h

```diff
@@ -83,6 +86,7 @@ class TrayMenu final : public QMenu
 
     bool m_omitPaste;
     bool m_viMode;
+    bool m_emacsMode;
```

Comment:
Can you change this to use an enum? Something like:

```cpp
enum class NavigationStyle {
    Default,
    Vi,
    Emacs
};
```

It would also mean that there is single method to set the navigation `void setNavigationStyle(NavigationStyle style)`.

It would be best if the configuration value in `appconfig.h` has this type, but that might require some more refactoring. I could look into this later.

Reply:
@hluk I looked into this today. One way I can see of making this change (let me know if this is not really what you had in mind) is to make `NavigationStyle` a `QVariant` and then bind `QGroupBox` instead of individual `QRadioButtons`.

I'd have to overload the `Option` constructor to allow me to pass getter and setter lambdas because `checkedId` is not available as a property. This overload would be used for `Option<NavigationStyle>` instance.

The main problem is this is backwards incompatible with persisted settings. For anyone that has `vi` mode enabled at the moment would reset to `Default` when they upgrade.

We can try to migrate settings, but unless we introduce a new concept (say `SettingsMigrator`) the implementation won't be clean.

Let me know if this isn't what you had in mind.

Reply:
I would prefer to keep the code simpler. I tried writing some conversion function for new enum option, but it did not work well and was unnecessarily complex (custom QMetaType conversion function).

Here is a simple solution: https://github.com/hluk/CopyQ/commit/28100299af198a9f412e86e9328481adb2a5c944

The UI uses QComboBox with fixed values instead of a group of radio buttons - enum values and the items in the combo box must match.

It reuses the `Config::vi` option because I thought that it would convert the old `false`/`true` values so it would stay backwards-compatible. Unfortunately, that does not work. It would be good to figure this out, but I would avoid any complex migration code.

Another downside is that from CLI (`copyq config vi ...`) you would need to use values 0,1,2 instead of more reasonable strings (e.g. "vi"/"emacs").

Reply:
I've solved the backwards-compatibility issue in: https://github.com/hluk/CopyQ/commit/aba654f6698e5ba16579c02bd2538415301f6def

But I'm not sure if I like the complexity of the solution 🙂 ([diff from the previous version](https://github.com/hluk/CopyQ/compare/28100299af198a9f412e86e9328481adb2a5c944..aba654f6698e5ba16579c02bd2538415301f6def)).

Reply:
@hluk you actually did all the work, I applied https://github.com/hluk/CopyQ/commit/aba654f6698e5ba16579c02bd2538415301f6def and doesn't look like there is anything left to do. 

I had a `predicate -> transform` arrangement in mind for migrating config over, that is, if `predicate` is true apply `transform`. So predicate would look for `vi=true`, remove it and replace with `navigation_style=vi` etc, but I have never worked with Qt before this, so not sure whether this would work well with `QSettings`.

If you prefer avoiding migration altogether, I can pull https://github.com/hluk/CopyQ/commit/28100299af198a9f412e86e9328481adb2a5c944 into this PR (and perhaps change it from `Config::vi` to `Config::navigation_style`).

What are the next steps you have in mind?